src/panels/lovelace/views/hui-view.ts

```diff
@@ -219,16 +226,47 @@ export class HUIView extends LitElement {
       configChanged = !oldLovelace || lovelace.config !== oldLovelace.config;
     }
 
+    let config = lovelace.config.views[this.index!];
+
+    if (!config.badges && !config.cards && (config.area || config.domain)) {
+      if (!subscribedRegistries) {
+        subscribedRegistries = true;
+        subscribeDeviceRegistry(hass.connection, () => undefined);
```

Comment:
```suggestion
        subscribeDeviceRegistry(hass.connection, (devices) => this.devices = devices);
```

Reply:
A subscribe function returns an unsubscribe function, you should store this and call it when you no longer want to listen (on disconnected for example) we have a mixin for handling all this, it is used in `ha-device-picker`: https://github.com/home-assistant/home-assistant-polymer/blob/54beaad7e51b0fd433e27ab0e3cd2fb5caf18449/src/components/device/ha-device-picker.ts#L43-L49